File: Leetcode_Hot_100/code/golang/01_two-sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, num := range nums {
		ans := target - num
		if _, ok := m[ans]; ok {
			return []int{i, m[ans]}
		} else {
			m[num] = i
		}
	}
	return nil
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9
	fmt.Println(twoSum(nums, target))
}

File: Leetcode_Hot_100/code/cpp/01_two-sum.cpp
// https://leetcode.cn/problems/two-sum

#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> m;
    for (int i = 0; i < nums.size(); i++)
    {
        int res = target - nums[i];
        auto iter = m.find(res);
        if (iter != m.end())
            return vector<int>{i, m[res]};
        m.insert(pair<int, int>(nums[i], i));
    }
    return vector<int>{-1, -1};
}

int main()
{
    vector<int> nums{ 2, 7, 11, 15 };
    int target = 9;
    auto ans = twoSum(nums, target);
    cout << ans[0] << " " << ans[1] << endl;
}
